### PR TITLE
platform(general): Catch None responses from BE

### PR DESCRIPTION
**Pull request by nimrodkor**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

We seldom get random connectivity issues which result in `None` value for the `message` field. This will handle it gracefully, retry 2 times and then exit gracefully

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/secrets/test_secret_git_history.py [hunk 1]
```diff
@@ -147,33 +147,8 @@ def test_scan_git_history_merge_added_removed2() -> None:
...
-    }) as settings:
-        settings.disable_filters(*['detect_secrets.filters.common.is_invalid_file'])
-        finished = GitHistoryScanner(valid_dir_path, secrets, None, 1).scan_history()
-
-    assert finished is False
-
-
 @mock.patch('checkov.secrets.scan_git_history.GitHistoryScanner._get_commits_diff', mock_run_forever)
-def test_scan_history_secrets_timeout2() -> None:
+def test_scan_history_secrets_timeout() -> None:
```
Comment (gruebel) [inline tests/secrets/test_secret_git_history.py:151]: 👍

**Review by gruebel — APPROVED** [review]

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -323,31 +323,42 @@ def get_s3_role(self, repo_id: str) -> tuple[str, str | None, dict[str, Any]]:
...
                                                 {"Authorization": token, "Content-Type": "application/json"},
                                                 get_user_agent_header()))
                 response = json.loads(request.data.decode("utf8"))
-
+            if response.get('message') is None and response.get('Message') is None:
+                if tries < 3:
+                    tries += 1
+                    response = self._get_s3_creds(repo_id, token)
+                else:
+                    raise BridgecrewAuthError("Checkov got an unexpected error that may be due to BE issues. Please contact support.")
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:346]:
```suggestion
                    raise BridgecrewAuthError("Checkov got an unexpected error that may be due to backend issues. Please contact support.")
```

File: checkov/common/bridgecrew/platform_integration.py [hunk 3]
```diff
@@ -323,31 +323,42 @@ def get_s3_role(self, repo_id: str) -> tuple[str, str | None, dict[str, Any]]:
...
+            if response.get('Message') and response['Message'] == UNAUTHORIZED_MESSAGE:
                 raise BridgecrewAuthError()
-            if 'message' in response and ASSUME_ROLE_UNUATHORIZED_MESSAGE in response['message']:
+            if response.get('message') and ASSUME_ROLE_UNUATHORIZED_MESSAGE in response['message']:
                 raise BridgecrewAuthError(
                     "Checkov got an unexpected authorization error that may not be due to your credentials. Please contact support.")
-            if 'message' in response and "cannot be found" in response['message']:
+            if response.get('message') and "cannot be found" in response['message']:
                 self.loading_output("creating role")
                 request = self.http.request("POST", self.integrations_api_url, body=json.dumps({"repoId": repo_id}),  # type:ignore[no-untyped-call]
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:336]: Should this use `_get_s3_creds`?

Comment (nimrodkor) [reply checkov/common/bridgecrew/platform_integration.py]: right, tnx

**Review by mikeurbanski1 — APPROVED** [review]

**Pull request merged**